code/game/machinery/doors/door.dm

```diff
@@ -483,5 +488,10 @@
 	zap_flags &= ~ZAP_OBJ_DAMAGE
 	. = ..()
 
+/// Signal proc for [COMSIG_ATOM_MAGICALLY_UNLOCKED]. Open up when someone casts knock.
+/obj/machinery/door/proc/on_magic_unlock(datum/source, obj/effect/proc_holder/spell/aoe_turf/knock/spell, mob/living/caster)
+	SIGNAL_HANDLER
+
+	INVOKE_ASYNC(src, .proc/open)
```

Comment:
Similarly to the other comment, moving magic_unlock up to an object level would be an idea for genericizing the behavior here, but that's totally out of scope for this PR 🤷